library/init/category/cont.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2018 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sebastian Ullrich
+
+The continuation monad transformer.
+-/
+
+prelude
+import init.category.alternative init.category.combinators init.category.lift
+universes u v w
+
+/-- An implementation of [ContT](https://hackage.haskell.org/package/mtl-2.2.2/docs/Control-Monad-Cont.html#t:ContT) -/
+structure cont_t (ρ : Type u) (m : Type u → Type v) (α : Type u) : Type (max u v) :=
+(run : (α → m ρ) → m ρ)
```

Comment:
Do you see an application for `cont_t`? 
Code based on `call_cc` is usually quite cryptic, but some systems provide a native and reasonably efficient implementation. This is not the case here. Code using this `call_cc` will be very inefficient since closures are created all over the place.
My point is do we need it in the core lib? It feels it is in the category: "we can do it, but please don't use it".

Reply:
cont has some very special properties, like not having any possible implementation of monad_functor, or it being a monad with `m` having to be one. It therefore felt incomplete to omit it, but you are right of course that it probably won't be very useful. Should we move it to a test?

Reply:
> Should we move it to a test?
A test would be great.
In the future, we can have an example folder where we can put this kind of code.